### PR TITLE
Issue 399 group components by depth

### PR DESCRIPTION
**Pull request by ged-odoo**

### CODE REVIEW

File: src/context.ts [hunk 1]
```diff
@@ -26,36 +26,52 @@ export class Context extends EventBus {
...
+    let i = 0;
+    while (i < iLen) {
+      let subs = [subscriptions[i]];
+      let depth = subs[0].owner.__owl__.depth;
+      i++;
+      while (i < iLen && subscriptions[i].owner.__owl__.depth === depth) {
+        subs.push(subscriptions[i]);
+        i++;
       }
+      const p = Promise.all(subs.map(s => s.callback.call(s.owner, rev)));
```
Comment (aab-odoo) [inline src/context.ts]: you removed the `isMounted` check, I don't know if it was intended.

File: src/context.ts [hunk 2]
```diff
@@ -26,36 +26,52 @@ export class Context extends EventBus {
...
-        await render;
+    const subscriptions = this.subscriptions.update || [];
+    let iLen = subscriptions.length;
+    let i = 0;
+    while (i < iLen) {
+      let subs = [subscriptions[i]];
+      let depth = subs[0].owner.__owl__.depth;
+      i++;
+      while (i < iLen && subscriptions[i].owner.__owl__.depth === depth) {
+        subs.push(subscriptions[i]);
```
Comment (aab-odoo) [inline src/context.ts]: do we have the guarantee that components with the same depth are 'contigu'?

File: tests/context.test.ts [hunk 3]
```diff
@@ -76,6 +76,82 @@ describe("Context", () => {
...
+
+    const app = new App(env);
+    await app.mount(fixture);
+    expect(fixture.innerHTML).toBe(
+      "<div><span>123</span><div><span>123</span><span>123</span></div></div>"
+    );
+    testContext.state.value = 321;
+    await nextTick();
+    expect(steps).toEqual(["render"]);
+    // expect(steps).toEqual(["updateprops", "updateprops"]);
```
Comment (aab-odoo) [inline tests/context.test.ts]: remove? or uncomment

File: src/context.ts [hunk 4]
```diff
@@ -26,36 +26,52 @@ export class Context extends EventBus {
     this.state = this.observer.observe(state);
   }
 
+  on(eventType: string, owner: any, callback: Callback) {
+    owner = owner || { __owl__: {} };
+    super.on(eventType, owner, callback);
+  }
```
Comment (aab-odoo) [inline src/context.ts]: i don't know about this

**Comment by ged-odoo** [comment]
i updated the PR

**Pull request merged**